schema/productconfig.rnc

```diff
@@ -0,0 +1,475 @@
+  element defaultblurb {
+    attribute lang { ds.type.lang },
+    ds.htmlblock+
+  }
+
+ds.blurb =
+  element blurb {
+    attribute lang { ds.type.lang },
+    ds.htmlblock+
+  }
+
+
+# DOCSETS
+
+ds.docset =
+  element docset {
+    attribute setid { ds.type.id },
+    ds.version,
+    (
+      (ds.builddocs,
+       ds.extralinks?) |
+      ds.extralinks
+    )
+  }
+
+ds.version =
+  element version {
+    text
+  }
+
+ds.builddocs =
+  element builddocs {
+    ds.git,
+    ds.defaultlanguage,
+    ds.otherlanguage*
+  }
+
+ds.git =
+  element git {
+    # git is currently the only thing supported but let's leave that open
+    ds.remote,
+    ds.checkoutdir
+  }
+
+ds.remote =
+  element remote {
+    ds.type.gitremote
+  }
+
+ds.checkoutdir =
+  element checkoutdir {
+    ds.type.dirname
+  }
+
+ds.defaultlanguage =
+  element language {
+    attribute lang { ds.type.lang },
+    attribute default { "1" | "true" },
+    ds.branch,
+    ds.subdir?,
+    ds.deliverable+
+  }
+ds.otherlanguage =
+  element language {
+    attribute lang { ds.type.lang },
+    attribute default { "0" | "false" }?,
+    ds.branch?,
+    ds.subdir?,
+    ds.untranslated?
+  }
+
+
+ds.branch =
+  element branch {
+    ds.type.branch
+  }
+
+ds.subdir =
+  element subdir {
+    ds.type.dirname
+  }
+
+ds.deliverable =
+  element deliverable {
+    # FIXME: implementation probably uses yes/no values, xsd:boolean supports 0/1 or true/false!
+    attribute remarks { xsd:boolean }?,
+    attribute draft { xsd:boolean }?,
+    attribute meta { xsd:boolean }?,
+    ds.dc,
+    ds.format,
+    ds.styleroot?,
+    ds.param*,
+    ds.subdeliverable*
+  }
+
+ds.dc =
+  element dc {
+    ds.type.dcfile
+  }
+
+ds.format =
+  element format {
+    ds.type.format
+  }
+
+ds.styleroot =
+  element styleroot {
+    # FIXME: I think this this could either be a absolute/relative path or a URI
+    # xsd:token is probably too generous but I don't care enough right now.
+    xsd:token
+  }
+
+ds.param =
+  element param {
+    attribute name { ds.type.xsltparam },
+    text
+  }
+
+ds.subdeliverable =
+  element subdeliverable {
+    ds.type.id
+  }
+
+ds.untranslated =
+  element untranslated {
+    ds.deliverablerestricted+
+  }
+
+ds.deliverablerestricted =
+  element deliverable {
+    ds.dc,
+    ds.subdeliverable*
+  }
+
+ds.extralinks =
+  element extralinks {
+    ds.link+
+  }
+
+ds.link =
+  element link {
+    attribute href { xsd:anyURI },
+    (ds.htmlinline |
+    text)*
+  }
+
+
+# SUPPORTED HTML SUBSET
+
+ds.p =
+  element p {
+    ds.htmlattr,
+    (ds.htmlinline |
+     text)*
+  }
+
+ds.div =
+  element div {
+    ds.htmlattr,
+    (ds.htmlinline |
+     ds.htmlblock |
+     text)*
+  }
+
+ds.pre =
+  element pre {
+    text*
+  }
+
+ds.ul =
+  element ul {
+    ds.li+
+  }
+
+ds.ol =
+  element ol {
+    ds.li+
+  }
+
+ds.li =
+  element li {
+    ds.htmlattr,
+    ds.htmlblock*
+  }
+
+ds.h1 =
+  element h1 {
+    ds.htmlattr,
+    (ds.htmlinline |
```

Comment:
Seems like the `ds.htmlattr, (ds.htmlinline | text)*` pattern is so common that you could separate them into another pattern. Let's say `ds.html.inline.content`.

Reply:
Thanks. That looks a bit better at least. Done.